diff --git a/src/config.mak b/src/config.mak
index 3599f24..dce7e80 100755
--- a/src/config.mak
+++ b/src/config.mak
@@ -31,19 +31,12 @@ SRCTIFF = \
 SRCTIFF  := $(addprefix libtiff/, $(SRCTIFF)) im_format_tiff.cpp
 INCLUDES += libtiff 
 
-ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-  SRCPNG = \
-      png.c       pngget.c    pngread.c   pngrutil.c  pngwtran.c  \
-      pngerror.c  pngmem.c    pngrio.c    pngset.c    pngwio.c    \
-      pngpread.c  pngrtran.c  pngtrans.c  pngwrite.c  pngwutil.c
-  SRCPNG := $(addprefix libpng/, $(SRCPNG)) im_format_png.cpp
-  INCLUDES += libpng
-else
-  # In Linux, use the installed files in the system libpng-dev
-  # If using GTK, then must use the same libpng they use
-  INCLUDES += /usr/include/libpng
-  SRCPNG := im_format_png.cpp
-endif
+SRCPNG = \
+    png.c       pngget.c    pngread.c   pngrutil.c  pngwtran.c  \
+    pngerror.c  pngmem.c    pngrio.c    pngset.c    pngwio.c    \
+    pngpread.c  pngrtran.c  pngtrans.c  pngwrite.c  pngwutil.c
+SRCPNG := $(addprefix libpng/, $(SRCPNG)) im_format_png.cpp
+INCLUDES += libpng
 
 SRCJPEG = \
     jcapimin.c  jcmarker.c  jdapimin.c  jdinput.c   jdtrans.c   \
